tests/lib/hardware/libvirt.py

```diff
@@ -140,6 +150,30 @@ def _backing_file_create(self):
         logger.info(f"node {self.name}: created qcow2 backing file under"
                     f"{self._snap_img_path}")
 
+    def _data_disk_count(self):
+        count = len(minidom.parseString(self._dom.XMLDesc()).
+                    getElementsByTagName('devices')[0].
+                    getElementsByTagName('disk'))
+        # We have two device by default - OS and Cloudinit seed
+        return count - 2
+
+    def add_data_disk(self, capacity='10G'):
+        _id = self._data_disk_count()
+        volume_name = f'data-{_id}'
+        block_device = f'vd{string.ascii_lowercase[_id + 1]}'
+        disk_path = os.path.join(self._workspace.working_dir,
+                                 f"{self.name}-{volume_name}.qcow2")
+        execute(f"qemu-img create -f qcow2 {disk_path} {capacity}")
+        logger.info(f'Created volume "{volume_name}" / size={capacity}')
+        disk = textwrap.dedent("""
+            <disk type='file' device='disk'>
+                <driver name='qemu' type='qcow2' cache='none'/>
+                <source file='%(disk_path)s'/>
+                <target dev='%(block_device)s' bus='virtio'/>
+            </disk>
+        """ % {"disk_path": disk_path, "block_device": block_device})
+        self._dom.attachDevice(disk)
```

Comment:
(cont.) we could do something like `self.disks.append()` here to keep track of them.

Reply:
Done